client/source/js/modules/model/calibration-ctrl.js

```diff
@@ -63,6 +66,59 @@ define(['./module', 'angular', 'underscore'], function (module, angular, _) {
       if ($scope.projectInfo.has_data){
         $scope.simulate();
       }
+
+      /** When the window or gridster element are resized */
+      $scope.$on('gridster-resized', resizeAllCharts);
+
+      $scope.gridsterOpts = {
+        columns: 16,
+        margins: [10, 10],
+        resizable: {
+          // item finished resizing
+          stop: function(event, $element, widget) {
+            /**
+             * Use the '.gridster-preview-holder' instead '.chart-container' like in resizeCharts(),
+             * because while resizing the '.chart-container' width/height changing and are not yet final
+             */
+            var previewHolder = $('.gridster-preview-holder');
+            var newWidth = previewHolder.width();
+            var newHeight = previewHolder.height();
+            var chart = $scope.charts[widget.index];
+            resizeChart(chart, newWidth, newHeight);
+            // shallow copy instead of deep copy with angular.copy
+            $scope.charts[widget.index] = _.clone(chart);
+          }
+        }
+      }
+    };
+
+    /**
+     * Resize all charts. The resizing is done after the gridster items are already rendered.
+     */
+    var resizeAllCharts = function() {
+      // avoid double resizing during the initial render
+      if(initialResize === true) { initialResize = false; return; }
+      $('.chart-container').each(function (index, chartContainer) {
+        var newWidth = $(chartContainer).outerWidth();
+        var newHeight = $(chartContainer).outerHeight();
+        // happens when removing the chart
+        if(index < $scope.charts.length) {
+          var chart = $scope.charts[index];
+          resizeChart(chart, newWidth, newHeight);
+        }
+      });
+      $scope.showCharts = true;
+      // shallow copy instead of deep copy with angular.copy
+      if($scope.charts.length > 0)
+        $scope.charts[0] = _.clone($scope.charts[0]);
+    };
+
+    var resizeChart = function(chart, newWidth, newHeight) {
+      var widthOffset = 20;
+      var heightOffset = chart.type == 'stackedAreaChart' ? 210 : 100;
+      chart.options.width = newWidth - widthOffset;
+      chart.options.height = newHeight - heightOffset;
+      chart.id = generateChartId();
```

Comment:
Math.random has a change of not being unique. You can use http://underscorejs.org/#uniqueId to be sure it's unique in the DOM.